### ISSUE TITLE
D417 Error for function docstrings with dashed lines

### ISSUE THREAD

**Issue by eronnen** [astral-sh/ruff#7250]
```
check.py:8:5: D417 Missing argument description in the docstring for `b`: `y`
Found 1 error.
```

caused by the following script:

```python
"""Test Module."""

import logging

logging.root.setLevel(logging.INFO)

def b(y: int) -> int:
    """Calculate square.

    Args:
    ----
        y: integer to square

    Returns:
    -------
        int: the result.
    """
    return y*y

logging.info(b(4))
```

With the following configuration:

```toml
[tool.ruff]
select = ["D407", "D417"]
```

**Comment by charliermarsh** [comment]
For reference, pydocstyle doesn't really support this either. pydocstyle doesn't flag this "by accident", because pydocstyle assumes that the docstring is a NumPy-style docstring, and then doesn't validate the "Args:" section (which isn't a valid NumPy-style section -- for NumPy-style docstrings, it would need to be "Parameters").

I don't mind supporting this specific case since it seems straightforward, but FWIW I don't really recommend mixing conventions like this since it can weird to unusual results.
